### DISCUSSION TITLE
Whisper on accelerated edge devices

### DISCUSSION THREAD

**Discussion by maxbbraun** [openai/whisper#1124] [Show and tell]
Hi all!

I'm sharing [`whisper-edge`](https://github.com/maxbbraun/whisper-edge), a project to bring Whisper inference to edge devices with ML accelerator hardware. It's mainly meant for real-time transcription from a microphone.

It currently works reasonably well for [NVIDIA Jetson Nano](https://github.com/maxbbraun/whisper-edge#jetson-nano) after some tweaks to the runtime and without any need to modify the models themselves. There should still be plenty of room for improvement to optimize latency and quality. Some ideas [here](https://github.com/maxbbraun/whisper-edge/issues/3).

I also started exploring [Google Coral](https://github.com/maxbbraun/whisper-edge#coral-edge-tpu), which requires additional surgery on the models. Potential next steps are outlined [here](https://github.com/maxbbraun/whisper-edge/issues/1).

Hope this is useful to some of you!

Max

**Comment by essiebes** [comment]
This is a really cool addition to the Whisper project @maxbbraun.

I will be able to test it out on the new Jetson Orin Nano soon.
I'm curious how performance is going to compare with the regular Jetson Nano. Not quite sure how I am supposed to interpret the numbers you listed in the [README](https://github.com/maxbbraun/whisper-edge#model). 

~10x in isolating meaning for everyone 10 recorded seconds it needs 1 second to process?

**Comment by maxbbraun** [comment]
Thanks! Looking forward to your results.

Yes, I was seeing inference performance of about 1 second per 10 seconds of audio to process. I say in isolation, because once I added in the (very unoptimized) overhead of running the recording loop, it dropped to about 10 seconds per 10 seconds.

**Comment by quasoft** [comment]
Hi @essiebes, sorry for the tag, but this Jetson device looks awesome and I am really curious how your tests with Jetson Orin Nano and whisper went. Like, what is the largest model size you managed to fit in its memory?

**Comment by tnhoang** [comment]
Hi @maxbbraun, I'd like to know how long does it take to transcribe a 30' audio?
It took me 1' to do it in Macbook pro M1 with Whispercpp and I want to improve that. I also interested in Google Coral as well.

**Reply by maxbbraun** [reply]
The goal with this implementation was mainly real-time transcription and you could probably be more efficient for offline. I think the base model ran at about 10x, so 3'.
